arduino-ide-extension/src/node/core-client-provider.ts

```diff
@@ -90,17 +93,17 @@ export class CoreClientProvider extends GrpcClientProvider<CoreClientProvider.Cl
           this._initialized.resolve();
           this.updateIndex(this._client); // Update the indexes asynchronously
         } catch (error: unknown) {
-          if (
-            this.isPackageIndexMissingError(error) ||
-            this.isDiscoveryNotFoundError(error)
-          ) {
+          console.error(
+            'Error occurred while initializing the core gRPC client provider',
+            error
+          );
+          if (error instanceof IndexUpdateRequiredBeforeInitError) {
             // If it's a first start, IDE2 must run index update before the init request.
             await this.updateIndexes(this._client);
             await this.initInstance(this._client);
             this._initialized.resolve();
-          } else {
-            throw error;
           }
+          throw error;
```

Comment:
Incorrect; I have just noticed. It should be in the `else` branch. It does not cause malfunction but is still lousy code.